File: Java/src/WordSearchTest.java
import org.junit.Assert;
import org.junit.Test;

public class WordSearchTest {

    @Test
    public void testExist() {
        WordSearch ws = new WordSearch();

        Assert.assertFalse(ws.exist(new char[][]{{}}, "AB"));

        Assert.assertFalse(ws.exist(null, "AB"));

        char[][] chars = new char[][]{
                {'A', 'B', 'C', 'E'},
                {'S', 'F', 'C', 'S'},
                {'A', 'D', 'E', 'E'}};

        Assert.assertTrue(ws.exist(chars, "ABCCED"));

        Assert.assertTrue(ws.exist(chars, "SEE"));

        Assert.assertFalse(ws.exist(chars, "ABCB"));
    }

}

File: Kotlin/src/WordSearchTest.kt
import org.junit.Test
import kotlin.test.assertFalse
import kotlin.test.assertTrue

class WordSearchTest {

    @Test
    fun testExist() {
        val ws = WordSearch()

        // Expected: false
        assertFalse(ws.exist(arrayOf(charArrayOf()), "AB"))

        val chars = arrayOf(charArrayOf('A', 'B', 'C', 'E'), charArrayOf('S', 'F', 'C', 'S'), charArrayOf('A', 'D', 'E', 'E'))

        assertTrue(ws.exist(chars, "ABCCED"))

        assertTrue(ws.exist(chars, "SEE"))

        assertFalse(ws.exist(chars, "ABCB"))
    }

}